krun/platform.py

```diff
@@ -696,8 +697,13 @@ def pin_process_args(self):
         return []  # not supported on OpenBSD
 
     def is_virtual(self):
-        """Not yet implemented on this platform, assume not virtualised"""
+        """If we see a vio(4) disk in the dmesg, this is a virtual machine"""
 
+        dmesg_lines = self._collect_dmesg_lines()
+        comp_pat = re.compile(OpenBSDPlatform.VIO_DMESG_PATTERN)
+        for line in dmesg_lines:
```

Comment:
Might be simpler as something like `return any(comp_pat.match(l) for l in dmesg_lines)`?

Reply:
Simpler to read, much less efficient. Thoughts?

Reply:
Is this a performance critical part of things? [I think not, but I might be wrong!]

Reply:
Well, not hugely, it's just a waste of cycles, and regex aint exactly efficient.

My dmesg is 528 lines long. That's a lot of regex.

Reply:
I'd rather have easy to read code: you'd be amazed at how quick that code will execute on a modern machine!

Reply:
I'm well aware of how fast a machine is :P 

I don't feel strongly enough to argue more :) Fixed.

Reply:
Oh, I see what you were implying. You needn't worry :)

```python
>>> type(x for x in [1])
<type 'generator'>
>>> type([x for x in [1]])
<type 'list'>
```

Reply:
Um. Not sure what you mean. I just didn't want to do load of regex and allocate a big list if it aint necessary. The vio entry, if it exists, will be early in the dmesg.

If it makes me look cleverer, i'll take credit for your code snippet though :)

Reply:
What I mean is that the version I'm supplying lazily evaluates the list, so its performance is the same as the `for` loop. So, I believe, it's smaller code that's (more or less) the same performance.

You're welcome to take the credit though -- you'll also get the blame ;)

Reply:
Ah, the generator doesn't allocate the list?